include/stl2/detail/memory/uninitialized_copy.hpp

```diff
@@ -0,0 +1,72 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Casey Carter 2016
+//  Copyright Christopher Di Bella 2016
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_DETAIL_ALGORITHM_UNINITIALIZED_COPY_HPP
+#define STL2_DETAIL_ALGORITHM_UNINITIALIZED_COPY_HPP
+
+#include <new>
+#include <stl2/iterator.hpp>
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/memory/destroy.hpp>
+#include <stl2/detail/memory/reference_to.hpp>
+#include <stl2/detail/tagged.hpp>
+
+STL2_OPEN_NAMESPACE {
+   ///////////////////////////////////////////////////////////////////////////
+   // uninitialized_copy [Extension]
+   //
+   template <InputIterator I, Sentinel<I> S, /*NoThrow*/ForwardIterator O>
```

Comment:
> Ahh, so it's a semantic difference!

Yes. We could feasibly *require* the operations to be `noexcept` with Concepts, but it would be a breaking change with Standard C++, and not a hugely valuable break at that. We decided that actually requiring `noexcept` for `Destructible` was not too breaking given that destructors are implicitly `noexcept` for most sane types.

> Should there be an exposition `__NoThrowForwardRange`?
`__NoThrowForwardRange<T>` is `Range<T>() && __NoThrowForwardIterator<iterator_t<T>>()`. If you prefer to type `__NoThrowForwardRange<T>`, then by all means define it. I have no preference.